src/js/components/FileInput/FileInput.js

```diff
@@ -149,6 +149,7 @@ const FileInput = forwardRef(
                 hoverIndicator
                 onClick={event => {
                   event.stopPropagation();
+                  if (onChange) onChange(event);
```

Comment:
I think this will send the click event from the button, which isn't going to return the same kind of event that the input will return, i.e. there is no `files` in the event.

Reply:
I added as a second argument the `files` from the hook, however, I'm not sure if it's the best option.
If you have any suggestion for this, let me know.

Reply:
I think this is the right strategy, to add an argument, as it parallels how grommet Select onChange works. To make it align even more, I think we should do: `onChange(event, { files })`. That will make it easier to extend in the future.

I also think we should change the onChange delivered from StyledFileInput to include the same second argument, that way the caller can always refer to the files in the same way.